scripts/egp39/previewProposal.ts

```diff
@@ -0,0 +1,306 @@
+  fs.writeFileSync("scripts/egp39/proposalArgs.json", data);
+}
+
+export async function getProposalArgs(
+  tokenAddress: string,
+  hdTokenAddress: string,
+  coreVotingAddress: string,
+  lockingVaultAddress: string,
+  vestingVaultAddress: string,
+  gscCoreVotingAddress: string,
+  gscVaultAddress: string,
+  hdLockingVaultAddress: string,
+  hdMigrationLinearVestingVaultAddress: string,
+  hdMigrationRewardsVaultAddress: string,
+  hdGscVaultAddress: string,
+  treasuryAddress: string,
+  timeLockAddress: string
+): Promise<ProposalArgs> {
+  // Actual proposal actions.
+  const elfiTokenInterface = new ethers.utils.Interface(
+    ERC20PermitWithMint__factory.abi
+  );
+
+  // Remove the LockingVault from the approved vault list in the CoreVoting contract.
+  const coreVotingInterface = new ethers.utils.Interface(
+    CoreVoting__factory.abi
+  );
+  const callDataChangeVaultStatusLockingVault =
+    coreVotingInterface.encodeFunctionData("changeVaultStatus", [
+      lockingVaultAddress,
+      false,
+    ]);
+
+  // Remove the VestingVault from the approved vault list in the CoreVoting contract.
+  const callDataChangeVaultStatusVestingVault =
+    coreVotingInterface.encodeFunctionData("changeVaultStatus", [
+      vestingVaultAddress,
+      false,
+    ]);
+
+  // Remove the gscVotingVault from the approved vault list in the gscCoreVoting contract.
+  const callDataChangeVaultStatusGscVault =
+    coreVotingInterface.encodeFunctionData("changeVaultStatus", [
+      gscVaultAddress,
+      false,
+    ]);
+
+  // Add the hdLockingVault to the approved vault list in the CoreVoting contract.
+  const callDataChangeVaultStatusHdLockingVault =
+    coreVotingInterface.encodeFunctionData("changeVaultStatus", [
+      hdLockingVaultAddress,
+      true,
+    ]);
+
+  // Add the hdMigrationRewardsVault to the approved vault list in the CoreVoting contract.
+  const callDataChangeVaultStatusHdMigrationRewardsVault =
+    coreVotingInterface.encodeFunctionData("changeVaultStatus", [
+      hdMigrationRewardsVaultAddress,
+      true,
+    ]);
+
+  // Add the hdMigrationLinearVestingVault to the approved vault list in the CoreVoting contract.
+  const callDataChangeVaultStatusHdMigrationLinearVestingVault =
+    coreVotingInterface.encodeFunctionData("changeVaultStatus", [
+      hdMigrationLinearVestingVaultAddress,
+      true,
+    ]);
+
+  // Add the hdLockingVault to the approved vault list in the gscCoreVoting contract.
+  const callDataChangeVaultStatusHdGscVault =
+    coreVotingInterface.encodeFunctionData("changeVaultStatus", [
+      hdGscVaultAddress,
+      true,
+    ]);
+
+  // Set approvals for the hdLockingVault from the treasury for HD tokens.
+  const treasuryInterface = new ethers.utils.Interface(Treasury__factory.abi);
+  console.log("hdTokenAddress", hdTokenAddress);
+  console.log("hdLockingVaultAddress", hdLockingVaultAddress);
+  const callDataTreasuryApproveHdLockingVault =
```

Comment:
We don't need to approve the locking vault or the GSC vault. We only need to approve the migration vaults because those are the vaults that have to pull HD from the treasury.